### PR TITLE
adding get all weets endpoint, adding new schema validation, adding p…

### PR DESCRIPTION
**Pull request by luis8alopez**
## Summary

- Adding new get all weets endpoint
- Protecting endpoint with jwt validation
- Protecting endpoint with schema validations
- Paginating response
- Adding test

## Trello Card

https://trello.com/c/ht4gXUrS/36-obtener-weets

### CODE REVIEW

File: app/services/weets.js [hunk 1]
```diff
@@ -34,3 +34,17 @@ exports.saveWeet = async (userId, content) => {
...
   }
 };
+
+exports.getAllWeets = async (page, weetLimit) => {
+  try {
+    let offset = 0;
+    offset += (page - 1) * weetLimit;
+    const allWeets = await Weet.findAll({ offset, limit: weetLimit });
+
+    const response = await allWeets.map(user => formatWeetOutput(user));
```
Comment (shernanm) [inline app/services/weets.js]: Here you can take into account what Dani commented in a previous PR.

File: app/controllers/weets.js [hunk 2]
```diff
@@ -14,3 +14,16 @@ exports.createWeet = async (req, res, next) => {
     return next(error);
   }
 };
+
+exports.getWeets = async (req, res, next) => {
+  try {
+    const { page, limit } = req.query;
+    logger.info('Fetching all weets from database');
+    const allWeets = await getAllWeets(page, limit);
+    const response = await allWeets.map(user => formatWeetOutput(user));
```
Comment (dbenavidesv) [inline app/controllers/weets.js]: You don't need the `await` here

File: test/weetsGetAll.test.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
+const request = require('supertest');
+const app = require('../app');
+const { signUpUser, pagination, badPagePagination, badLimitPagination } = require('./data/weets');
+
+describe('GET /weets - Return All weets created', () => {
+  /*eslint-disable */
```
Comment (dbenavidesv) [inline test/weetsGetAll.test.js]: Disable specific rules

File: test/weetsGetAll.test.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+      expect(body.length).toBe(2);
+    });
+  
+    it('Should return status code 200', () => {
+      expect(status).toBe(200);
+    });
+  });
+
+    describe('GET /weets - Unauthorized User', () => {
+      /*eslint-disable */
```
Comment (dbenavidesv) [inline test/weetsGetAll.test.js]: same

**Review by dbenavidesv — DISMISSED** [review]

**Review by shernanm — DISMISSED** [review]

**Review by shernanm — APPROVED** [review]

**Review by dbenavidesv — APPROVED** [review]

**Pull request merged**